Python/ceval_gil.c

```diff
@@ -452,9 +459,16 @@ static inline int
 current_thread_holds_gil(struct _gil_runtime_state *gil, PyThreadState *tstate)
 {
     if (((PyThreadState*)_Py_atomic_load_ptr_relaxed(&gil->last_holder)) != tstate) {
+#ifdef Py_GIL_DISABLED
```

Comment:
If we have `tstate->_status.holds_gil` in the default build, this function can just be `return tstate->_status.holds_gil`, right?

That would also avoid a race condition here where the GIL is acquired by another thread between the `last_holder` check and the `locked` check.

Reply:
> If we have `tstate->_status.holds_gil` in the default build, this function can just be `return tstate->_status.holds_gil`, right?

Yep. I'll probably keep an assert or two as a sanity check.